### PR TITLE
WIP: consolidated branch for feature developed in mta/parallel-batches

### PR DESCRIPTION
**Pull request by martetassyns**
Separate pull request to run automated tests (see related pull request for mta/parallel-batches-preview
(changes: based on stable 11.2.2 release instead of main, merged all small incremental changes up to 51079bb in a single commit)

### CODE REVIEW

**Comment by martetassyns** [comment]
@michaelstaib I'd like to use this pull request to run the automated tests on the exact version we made our local built from. Can you set status so that HotChocolate-CI and HotChocolate-Sonar are ran?

File: src/HotChocolate/Core/src/Execution/Processing/DefaultTransactionScopeHandler.cs [hunk 1]
```diff
@@ -27,7 +27,8 @@ public virtual ITransactionScope Create(IRequestContext context)
                     new TransactionOptions
                     {
                         IsolationLevel = IsolationLevel.ReadCommitted
-                    }));
+                    },
+                    TransactionScopeAsyncFlowOption.Enabled));
```
Comment (michaelstaib) [inline src/HotChocolate/Core/src/Execution/Processing/DefaultTransactionScopeHandler.cs:31]: This is correct but you should make a separate PR for this. We can merge this change then into 11.3

**Comment by martetassyns** [comment]
@michaelstaib I see that this pull request is closed now, does this mean that this functionality is now present in the 12 release?
If not, if I rewrite it for the 12 release, will it be accepted in the actual product?

**Comment by michaelstaib** [comment]
@martetassyns we tried to reach you but could not get a response on slack. We have now the query plan implemented so if we want to get a more efficient batching it needs to be build on top of that. If I remember right the main thing you wanted to achieve is executing batches efficiently in parallel.

**Comment by martetassyns** [comment]
My apologies, because I don't often use slack I only see messages when I login through the website.
That is correct, and we still require this functionality.
Is this currently on your own roadmap?
And if it would be implemented by me, do you foresee any blockers for an actual merge into the main branch?

**Comment by michaelstaib** [comment]
The functionality will be coming in V13 we are working on batching for stitching. I think I outlined back when we were talking that I would like to merge the queries to send them in as one query .... this is still what I want to do. With the new query plans we can finally do this by rewriting the selection types to nullables. This essentially will allow us to much more efficiently execute multiple requests as one request with all the benefits of the new execution engine.

If you want to help on that I am all ears. We can get together on a call next week if you like to.

**Comment by martetassyns** [comment]
Ok, thank you for the feedback.
I'll try to take a look at the current code and see if I can find my way in it.
